runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -333,11 +367,23 @@ impl<T: Config> Pallet<T> {
 			// Note that `provide_multi_dispute_data` will iterate, verify, and import each
 			// dispute; so the input here must be reasonably bounded.
 			let _ = T::DisputesHandler::provide_multi_dispute_data(disputes.clone())?;
+			dispute_sets_processed_metric
+				.with_label_values(&vec!["imported)"])
+				.inc_by(disputes.len() as u64);
+
 			if T::DisputesHandler::is_frozen() {
+				// Relay chain freeze, at this point we will not include any parachain blocks.
+				dispute_sets_processed_metric.with_label_values(&vec!["frozen)"]).inc_by(1);
+
 				// The relay chain we are currently on is invalid. Proceed no further on parachains.
 				return Ok(Some(dispute_statements_weight::<T>(&disputes)).into())
 			}
 
+			// Process the dispute sets of the current session.
+			dispute_sets_processed_metric
+				.with_label_values(&vec!["current_session)"])
```

Comment:
we only increment the counter, so current_session will track all sessions?

Reply:
Yes, it will track all sessions. I'm unsure this is useful, I guess it depends on how long is one session ?